File: src/main/java/jaba/adt/Point.java
package jaba.adt;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

/**
 * Created by Administrador on 21/05/2017.
 */
@Getter
@Setter
@AllArgsConstructor
public class Point {
    private double y;
    private double x;

    public Point() {
        x = Math.random();
        y = Math.random();
    }

    double r() {
        return Math.sqrt((x * x) + (y * y));
    }

    double theta() {
        return StrictMath.atan2(x, y);
    }

    double distance(Point point) {
        double dx = x - point.getX();
        double dy = y - point.getY();
        return Math.sqrt((dx * dx) + (dy * dy));
    }

    @Override
    public String toString() {
        return "(" + x + ", " + y + ')';
    }
}

File: src/main/java/katlin/adt/Point.kt
package katlin.adt

open class Point {
    open var y: Double
    open var x: Double

    constructor() {
        x = Math.random()
        y = Math.random()
    }

    constructor(y: Double, x: Double) {
        this.y = y
        this.x = x
    }

    internal fun r(): Double {
        return Math.sqrt(x * x + y * y)
    }

    internal fun theta(): Double {
        return Math.atan2(x, y)
    }

    internal fun distance(point: Point): Double {
        val dx = x - point.x
        val dy = y - point.y
        return Math.sqrt(dx * dx + dy * dy)
    }

    /**Returns a string representation of the object. */
    override fun toString(): String {
        return "($x, $y)"
    }
}